### PR TITLE
Bugfix: GUI: Options: Initialise prune setting range before loading current value, and remove upper bound limit

### PR DESCRIPTION
**Pull request by luke-jr**
This fixes two bugs:

1. The prune setting range was set *after* loading the current value. If users had a prune of (eg) 200, it would get limited to 99 before the range was raised. This is fixed by setting the range first.
2. The prune setting was limited to <= the chainparams' "assumed blockchain size". There's no reason for this limit (the UX is the same either way), and there are use cases it breaks (eg, setting a prune size such that it begins pruning at some future point). Therefore, I raised it to the max value.

This is a daggy fix, so should cleanly merge to both master and 0.18 branches.

### CODE REVIEW

File: src/qt/optionsdialog.cpp [hunk 1]
```diff
@@ -148,6 +148,10 @@ void OptionsDialog::setModel(OptionsModel *_model)
         if (_model->isRestartRequired())
             showRestartWarning(true);
 
+        // Prune values are in GB to be consistent with intro.cpp
+        static constexpr uint64_t nMinDiskSpace = (MIN_DISK_SPACE_FOR_BLOCK_FILES / GB_BYTES) + (MIN_DISK_SPACE_FOR_BLOCK_FILES % GB_BYTES) ? 1 : 0;
+        ui->pruneSize->setRange(nMinDiskSpace, std::numeric_limits<int>::max());
```
Comment (MarcoFalke) [inline src/qt/optionsdialog.cpp]: Why was this changed in the first place?

Couldn't you restore the `ui->pruneSize->setMinimum(nMinDiskSpace);` like it was before?

Comment (luke-jr) [reply src/qt/optionsdialog.cpp]: The default max is 99.

**Comment by MarcoFalke** [comment]
utACK 8a33f4d63f9944f4877b3e2814b1582e72ceaa71

**Comment by luke-jr** [comment]
(should probably get tagged for 0.18 since we're doing another rc)

File: src/qt/optionsdialog.cpp (hunk 1 above)
Comment (promag) [reply src/qt/optionsdialog.cpp]: ACK, doc is here: https://doc.qt.io/qt-5/qspinbox.html#maximum-prop.

**Review by promag — COMMENTED** [review]
utACK 8a33f4d.

**Comment by laanwj** [comment]
utACK 8a33f4d63f9944f4877b3e2814b1582e72ceaa71

**Comment by goomario** [comment]
Look like QSpinBox::setRange() not effect on Qt5.9.8 macOS Catalina.
